lucene/spatial-extras/src/java/org/apache/lucene/spatial/spatial4j/Geo3dDistanceCalculator.java

```diff
@@ -70,7 +71,22 @@ public boolean within(Point from, double toX, double toY, double distance) {
 
   @Override
   public Point pointOnBearing(Point from, double distDEG, double bearingDEG, SpatialContext ctx, Point reuse) {
-    throw new UnsupportedOperationException();
+    double dist = DistanceUtils.DEGREES_TO_RADIANS * distDEG;
+    double bearing = DistanceUtils.DEGREES_TO_RADIANS * bearingDEG;
+    Geo3dPointShape geoFrom = (Geo3dPointShape) from;
+    GeoPoint point = (GeoPoint)geoFrom.shape;
+
```

Comment:
Can you please add the name of this formula (if there is one) and/or a URL for further info?

Reply:
I have changes the formula to use vincenty's formulae. Slower as it is iterative but It taken into account the planet model.